pkgs/applications/office/jabref/default.nix

```diff
@@ -106,7 +106,7 @@ stdenv.mkDerivation rec {
     unzip $out/lib/javafx-web-17.0.1-linux.jar libjfxwebkit.so -d $out/lib/
 
     wrapProgram $out/bin/JabRef \
-      --prefix PATH : ${lib.makeBinPath [ xdg_utils ]} \
+      --prefix PATH : ${lib.makeBinPath ([] ++ lib.optionals stdenv.isLinux [ xdg_utils ])} \
```

Comment:
doesn't need to be conditional

Reply:
The CI failed without this. I guess there id no xdg-utils in OSX.

Reply:
Maybe I should disable other arches. I can't test them and it seems the deps hash would be different.

Reply:
The ci failed because aliases aren't allowed in nixpkg

The reason why ci now passed is because xdg_utils now points to null instead of a alias

Reply:
Thanks!

Reply:
failed on aarch64-linux

```
hash mismatch in fixed-output derivation '/nix/store/xbjz0lgg8aqrp508yn84zgbxl92yh4wc-jabref-deps-5.5':
  wanted: sha256:1k56wcjxxmfm3ghwamyyw1ris3qnqhi6xaw1qzqbfnvzz3bwa12m
  got:    sha256:0054qdaq2cwpvjg96hv88imfh9ppjqcm23k2wgah411f5rc9blzn
```

Reply:
also failed to build on x86_64-linux
```
hash mismatch in fixed-output derivation '/nix/store/729ia612fkbrjqasy3qgkjrqryraxr6w-jabref-deps-5.5':
  wanted: sha256:1k56wcjxxmfm3ghwamyyw1ris3qnqhi6xaw1qzqbfnvzz3bwa12m
  got:    sha256:1px841hd64lg7z93yvm3f0msi88zfcsvpbr657ppvrjkz4f9yvjy
```

so there seems to be some impurity ..

Reply:
> maybe make deps a function to generate one deps-derivation per platform (by mapping it over a list of platform-hash-tuples)

I'l add hashes per platform. Thanks!

> so there seems to be some impurity ..

What should I do in such a case...

Reply:
> > so there seems to be some impurity ..
> 
> What should I do in such a case...

This seems to be the culprit. You need to produce some sort of lockfile for the dependencies.
There's either gradle2nix or you can try an approach similar to the one used at status.im:
https://drive.google.com/file/d/1Ti0wppMoj40icCPdHy7mJcQj__DeaYBE/view?usp=sharing slide 20 (this is linked from https://our.status.im/reproducable-builds-with-nix/ where you can also find the link to the accompanying youtube vid).
The resulting nix/json/whatever file with unique identifiers of Gradle deps needs then be added to nixpkgs.

Reply:
I use gradle.lockfile for dependency lock. How can I got the dependencies for other platform on x86-64?